### PR TITLE
Format sources and add type checker

### PR DESCRIPTION
**Pull request by kaklakariada**

### CODE REVIEW

File: tools/fetch-lua-type-definitions.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+base_dir="$( cd "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P )"
```
Comment (tomuben) [inline tools/fetch-lua-type-definitions.sh:7]: You do permanently change the directory here, don't you?
Consider using `pushd` and later `popd`

File: tools/run-type-check.sh [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+base_dir="$( cd "$(dirname "$0")/.." >/dev/null 2>&1 ; pwd -P )"
```
Comment (tomuben) [inline tools/run-type-check.sh:7]: same as above

File: tools/run-type-check.sh
Comment (tomuben) [inline tools/run-type-check.sh:1]: I would recommend to run `spellcheck` in the CI.....

**Review by tomuben — APPROVED** [review]

File: tools/run-type-check.sh
Comment (kaklakariada) [reply tools/run-type-check.sh:1]: ✅

**Review by tomuben — APPROVED** [review]

**Pull request merged**